server/src/main/java/io/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -202,7 +226,8 @@ public void cleanupCacheFiles(File baseFile, File cacheFile) throws IOException
       }
     }
 
-    if (cacheFile.getParentFile() != null && cacheFile.getParentFile().listFiles().length == 0) {
+    if (cacheFile.getParentFile() != null &&
+        (cacheFile.getParentFile().listFiles() == null || cacheFile.getParentFile().listFiles().length == 0)) {
```

Comment:
Maybe extract `cacheFile.getParentFile().listFiles()` as a local variable? This is an expensive operation.